implementations/rust/ockam/ockam_command/src/identity/show.rs

```diff
@@ -62,16 +66,16 @@ impl ShowCommand {
             if Some(EncodeFormat::Hex) == cmd.encoding {
                 opts.println(&hex::encode(change_history.export().into_diagnostic()?))?;
             } else {
-                let identity = Identity::import_from_change_history(
+                let identity: ShowIdentity = Identity::import_from_change_history(
                     Some(&identifier),
                     change_history,
                     Vault::create_verifying_vault(),
                 )
                 .await
-                .into_diagnostic()?;
+                .into_diagnostic()?
+                .into();
 
-                let identity_display = IdentityDisplay(identity);
-                opts.println(&identity_display)?;
+                opts.println(&identity)?;
```

Comment:
While we are here, perhaps it's a good time to refactor this a little. The `opts.println` is a workaround that we are slowly deprecating in favor of `opts.terminal.stdout()...` like in:

https://github.com/build-trust/ockam/blob/656f03e53415267ae9017f37dca8f44a04113b95/implementations/rust/ockam/ockam_command/src/identity/create.rs#L131-L149

If you can do that in this PR it would be great 🙏

Reply:
And there is also a potential refactor around the `if cmd.full` block. Perhaps you could return the `plain, json` contents like `let (plain, json) = if cmd.full { ... }` so you only have to go through the `opts.terminal.stdout()` once.

Reply:
I made those changes. I also added json output for non-full and `--encoding hex --full` variants as that was easier and better that running the JSON conditionally. Once you know which names we should use I will make the last updates.

Reply:
We don't have an agreement yet so I'm going to mark this as resolved and continue with the review.